packages/grid/x-grid/src/XGrid.test.tsx

```diff
@@ -258,4 +220,48 @@ describe('<XGrid />', () => {
       expect(getColumnValues()).to.deep.equal(['Puma', 'Nike', 'Adidas']);
     });
   });
+  describe('state', () => {
+    it('should trigger on state change and pass the correct params', () => {
+      let onStateParams;
+      let apiRef;
+      function Test() {
+        apiRef = useApiRef();
+        const onStateChange = (params) => {
+          onStateParams = params;
+        };
+
+        return (
+          <div style={{ width: 300, height: 300 }}>
+            <XGrid {...defaultProps} onStateChange={onStateChange} apiRef={apiRef} />
+          </div>
+        );
+      }
+      render(<Test />);
+      const header = screen.getByRole('columnheader', { name: 'brand' });
+      fireEvent.click(header);
+      expect(onStateParams.api).to.eq(apiRef.current);
+      expect(onStateParams.state).to.eq(apiRef.current.state);
```

Comment:
`eq` is not allowed

```suggestion
      expect(onStateParams.api).to.equal(apiRef.current);
```

x2